aries_vcx/src/handlers/issuance/credential_def/revocation_registry.rs

```diff
@@ -0,0 +1,61 @@
+use crate::error::{VcxErrorKind, VcxResult};
+use crate::handlers::issuance::credential_def::PublicEntityStateType;
+use crate::libindy::utils::anoncreds;
+use crate::libindy::utils::anoncreds::RevocationRegistryDefinition;
+
+#[derive(Clone, Deserialize, Debug, Serialize, PartialEq)]
+pub struct RevocationRegistry {
+    pub rev_reg_id: String,
+    pub(super) rev_reg_def: RevocationRegistryDefinition,
+    pub(super) rev_reg_entry: String,
+    pub(super) tails_dir: String,
+    pub(super) max_creds: u32,
+    pub(super) tag: u32,
+    state_rev_reg_def: PublicEntityStateType,
+    state_rev_reg_delta: PublicEntityStateType,
+}
+
+
+impl RevocationRegistry {
+    pub fn create(issuer_did: &str, cred_def_id: &str, tails_dir: &str, max_creds: u32, tag_no: u32) -> VcxResult<RevocationRegistry> { // issuer_did: &str, cred_def_id: &str, tails_dir: &str, max_creds: u32, tag_no: u32
+        trace!("RevocationRegistry::create >>> issuer_did: {}, cred_def_id: {}, tails_dir: {}, max_creds: {}, tag_no: {}",  issuer_did, cred_def_id, tails_dir, max_creds, tag_no);
+        let (rev_reg_id, rev_reg_def, rev_reg_entry) =
+            anoncreds::generate_rev_reg(&issuer_did, &cred_def_id, tails_dir, max_creds, &format!("tag{}", tag_no))
+                .map_err(|err| err.map(VcxErrorKind::CreateRevRegDef, "Cannot create Revocation Registry"))?;
+        Ok(RevocationRegistry {
+            rev_reg_id,
+            rev_reg_def,
+            rev_reg_entry,
+            tails_dir: tails_dir.to_string(),
+            max_creds,
+            tag: tag_no,
+            state_rev_reg_def: PublicEntityStateType::Built,
+            state_rev_reg_delta: PublicEntityStateType::Built,
+        })
+    }
+
+    pub fn was_rev_reg_def_published(&self) -> bool {
+        self.state_rev_reg_def == PublicEntityStateType::Published
+    }
+
+    pub fn was_rev_reg_delta_published(&self) -> bool {
+        self.state_rev_reg_delta == PublicEntityStateType::Published
+    }
+
+    pub fn publish_rev_reg_def(&mut self, issuer_did: &str, tails_url: &str) -> VcxResult<()> {
+        trace!("RevocationRegistry::publish_rev_reg_def >>> Publish revocation registry. issuer_did:{}, rev_reg_id: {}, rev_reg_def:{:?}", issuer_did, &self.rev_reg_id, &self.rev_reg_def);
```

Comment:
Missing spaces. The comments after entrypoint, like "Publish revocation registry." are inconsistent and distracting